src/test/java/net/minecraftforge/test/MiningDimensionTest.java

```diff
@@ -0,0 +1,640 @@
+import net.minecraftforge.fml.common.FMLCommonHandler;
+import net.minecraftforge.fml.common.Mod;
+import net.minecraftforge.fml.common.Mod.EventHandler;
+import net.minecraftforge.fml.common.Mod.Instance;
+import net.minecraftforge.fml.common.event.FMLInitializationEvent;
+import net.minecraftforge.fml.common.event.FMLPostInitializationEvent;
+import net.minecraftforge.fml.common.event.FMLServerStartingEvent;
+import net.minecraftforge.fml.common.eventhandler.EventPriority;
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
+import net.minecraftforge.fml.common.network.FMLEmbeddedChannel;
+import net.minecraftforge.fml.common.network.FMLNetworkEvent;
+import net.minecraftforge.fml.common.network.FMLOutboundHandler;
+import net.minecraftforge.fml.common.network.NetworkRegistry;
+import net.minecraftforge.fml.common.network.handshake.NetworkDispatcher;
+import net.minecraftforge.fml.relauncher.Side;
+
+@Mod(
+        modid = "miningdimensiontest",
+        name = "MiningDimensionTest",
+        version = "1.0.0",
+        acceptableRemoteVersions = "*"/*, serverSideOnly = true*/)
+public class MiningDimensionTest
+{
+
+    public static final String MOD_NAME = "MiningDimensionTest";
+    public static final String MOD_ID = "miningdimensiontest";
+    public static final String commandName = "gmining";
+    public static String dimtypename = "miningdimensiontest";
+    @Instance("miningdimensiontest")
+    public static MiningDimensionTest inst;
+    public static DimensionType dimtypetest;
+    public static int dimtypetestid;
+    public static boolean keepinv = true;
+    public static boolean enableMapFeatures = false;// bug if true: Caused by: java.lang.NullPointerException
+    // net.minecraft.world.gen.feature.WorldGenFossils.generate(WorldGenFossils.java:45)
+    public static WorldType worldtype;
+    public static String savefoldername;
+    public static GameType gameType;
+    public static int miningid;// this should be saved on a file or something
+    static
+    {
+        dimtypetest = DimensionType.OVERWORLD;
+        worldtype = WorldType.AMPLIFIED;
+        savefoldername = "mining";
+        gameType = GameType.SURVIVAL;
+    }
+
+    @EventHandler
+    public void onInit(FMLInitializationEvent event)
+    {
+        // dimtypetestid = getNextFreeDimentionTypeId();
+        // dimtypetest = DimensionType.register(dimtypename, "_mine", dimtypetestid , WorldProviderTest.class, false); //does not work... DimensionType... getConstructor(new
+        // Class[0]) NoSuchMethodException
+        // System.out.println("MiningDimensionTest Registered dimension type "+ dimtypetest + " "+ dimtypetest.name() + " " + dimtypetest.getId());
+
+    }
+
+    @EventHandler
+    public void onServerStarting(FMLServerStartingEvent event)
+    {
+        event.registerServerCommand(new MiningDimensionTest.DimensionTestCommand());
+        System.out.println("MiningDimensionTest Registered  command /mining <int>");
```

Comment:
Please use the logger you get from `FMLPreInitializationEvent#getModLog` instead of `System.out`